File: Java/Factorial.java
import java.util.Scanner;

public class Factorial {
  static int facto = 1;
  public static void main(String[] args) {

    Scanner scanner = new Scanner(System.in);
    int numero = 0;

    System.out.print("Dime numero y te digo su factorial: ");
    numero = scanner.nextInt();
    System.out.println("El factorial es: " + dimeFactorial(numero));
    scanner.close();
  }
  
  static int dimeFactorial(int num) {
    facto *= num;
    return ( num == 1 ) ? facto : dimeFactorial(--num);  
  }
}

File: C/Factorial.c
#include <stdio.h>

int factorial (int);

int main() {
  int num;
  int resultado;
  printf("Introduzca un numero entero: ");
  scanf("%d", &num);
  resultado = factorial(num);
  printf("El factorial de %d es %d\n", num, resultado);
}

int factorial(int n) {
  if (n > 1) {return n*factorial(n-1);}
  else return 1;
}